src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,145 @@
 package core.basesyntax;
 
+import static java.util.Objects.hash;
+
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int MAXIMUM_CAPACITY = 1073741824;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final int GROW_CAPACITY_FACTOR = 2;
+
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
+
     @Override
     public void put(K key, V value) {
-
+        putVal(hash(key), key, value);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> node = getNode(key);
+        if (node == null) {
+            return null;
+        } else {
+            return node.value;
+        }
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putVal(int hash, K key, V value) {
+        Node<K, V> newNode = new Node<>(hash, key, value, null);
+        int index = getIndexFromHash(hash);
+
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> node = table[index];
+            do {
+                if (node.hash == hash(key) && Objects.equals(key, node.key)) {
```

Comment:
Using `hash(key)` again inside the if condition is redundant since you already have the hash passed as a parameter to `putVal` method. Use the `hash` parameter directly to avoid recomputing the hash.